spec/spec_helper.rb

```diff
@@ -9,36 +9,40 @@ module Adder
     defprotocol do
       defmethod :add, :this, :other
       defmethod :subtract, :this, :other
+      defmethod :to_s, :this
 
       def add_default(value)
         add(3, 2) + value
       end
     end
 
-    defimpl Protocols::Adder, for: String do
+    defimpl Protocols::Adder, target: String do
       def add(this, other)
         this * other
       end
-      def subtract(this, other)
+
+      def subtract(this, _other)
         this
       end
+
+      def to_s(this)
+        this.to_s
+      end
     end
-    defimpl Protocols::Adder, for: NilClass do
-      def add(this, other)
+    defimpl Protocols::Adder, target: NilClass do
```

Comment:
It would be cool (just because) to have a `defimpl` call outside. I'm not sure if it has to be a bare call to a `Kernel`-included method, or a properly namespaced one (to avoid polluting global namespace):
```ruby
# somewhere.rb
Dry::Protocol.defimpl Protocols::Adder, target: NilClass do
  ...
end
```


In any case implementations defined in the protocol itself shouldn't need the namespace (the one for `Integer` in this example)